verbose_debugging_reports.md

```diff
@@ -1,175 +1,282 @@
 Verbose Debugging Reports
 =========================
 
-This document is a collection of [verbose debugging
-reports](https://github.com/WICG/attribution-reporting-api/blob/main/EVENT.md#verbose-debugging-reports)
-that are supported.
+This document describes the format of [verbose debugging reports][].
+
+The report data is included in the request body as a JSON list of dictionaries.
+Each dictionary has a string field `type` and a dictionary field `body`, e.g.:
+
+```json
+[{
+  "type": "source-destination-limit",
+  "body": {
+    "attribution_destination": "https://destination.example",
+    "limit": "100",
+    "source_debug_key": "1234567890",
+    "source_event_id": "12340873456",
+    "source_site": "https://source.example"
+  }
+}]
+```
+
+The content of the `body` field depends on the `type`.
 
 ### Source debugging reports
 
-Here are the debugging reports supported for [attribution source
-registrations](https://github.com/WICG/attribution-reporting-api/blob/main/EVENT.md#registering-attribution-sources).
+The following reports are produced in response to [attribution source
+registrations][].
+
+The `body` will contain the following fields for all source debugging reports:
+
+* `attribution_destination`: The source registration's parsed `destination`
+  sites (i.e. with URLs replaced with sites, duplicates removed, and sorted).
+  This will be a string if there was one such site, or a list of strings if
+  there were multiple.
+* `source_event_id`: The source registration's `source_event_id`.
+* `source_site`: The top-level site on which the source registration occurred.
+
+Additionally:
+
+* If the source registration contained a valid `debug_key` and [cookie-based
+  debugging][] was allowed, then the `body` will also contain a
+  `source_debug_key` field.
+* If the source registration was rejected due to an API limit, then the `body`
+  will also contain a string-typed `limit` field.
 
 #### `source-destination-limit`
-A source is rejected due to the [destination limit](https://github.com/WICG/attribution-reporting-api/blob/main/EVENT.md#limiting-the-number-of-unique-destinations-covered-by-unexpired-sources).
+
+The source was rejected due to the [destination limit][].
+
+Additional fields: `limit`
 
 #### `source-noised`
-[Noise](https://github.com/WICG/attribution-reporting-api/blob/main/EVENT.md#data-limits-and-noise) is applied to a source event.
+
+The source was successfully registered, but it will not be attributable by any
+subsequent trigger because [noise][] has been applied.
+
+The `body` may also include a `source_destination_limit` field if the
+[destination limit][] was exceeded.
+
+Issue: It is unclear where in the [specification][] the
```

Comment:
It's added when `source-destination-limit-replaced` is included in the data types to report, which happens when the source destination limit is hit and deleting old sources.

Reply:
Would you mind providing a link to the relevant step? IIUC [this](https://wicg.github.io/attribution-reporting-api/#ref-for-source-debug-data-type-source-noised) is the only step that produces a `source-noised` value.

Reply:
Ah. I missed the fact that we are modifying `body` in place by looping over multiple data types. I think that's fairly confusing and could do with some assertions, at the very least, if not a general refactor to make it more clear what's going on.